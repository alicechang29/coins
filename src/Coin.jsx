import React from "react";
// import "./Coin.css";

/** Coin: displays heads or tails
 *
 * Props:
 * - caption: string describing the coin side: Heads or Tails
 * - src: string for the image link
 *
 * State:
 * - none
 *
 * App --> Game --> Coin
 */

function Coin({ caption = "", src = "" }) {

  return (
    <div className="Coin">
      <h4 className="Coin-title">{caption}</h4>
      <img className="Coin-image" src={src} alt={caption}></img>
    </div>
  );
}


export default Coin;