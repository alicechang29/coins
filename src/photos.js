
const HEADS = "https://coinsofamerica.com/wp-content/uploads/state-quarter-d-mint.jpg";
const TAILS = "https://www.greatamericancoincompany.com/cdn/shop/products/roll-of-40-90-silver-washington-quarters-10-face-bu-metal-united-states-mint-great-american-coin-companyr_219_480x480.jpg?v=1569285048";

const coins = [
  {
    src: HEADS,
    caption: "Heads"
  },
  {
    src: TAILS,
    caption: "Tails"
  }
];

export { HEADS, TAILS };
export default coins;