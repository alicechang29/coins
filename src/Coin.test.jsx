import { describe, it, expect } from "vitest";
import React from "react";
import { render } from "@testing-library/react";
import Coin from "./Coin.jsx";
import { HEADS } from "./photos.js";

it("renders without crashing", function () {
  // this is a low-value test, but better than nothing
  render(<Coin
    caption="Heads"
    src={HEADS}
  />);
});

it("has the correct alt text & src", function () {
  const { container, debug } = render(<Coin
    caption="Heads"
    src={HEADS}
  />);
  const img = container.querySelector("img");
  debug(img);

  expect(img.getAttribute("alt"))
    .toEqual("Heads");
  expect(img.getAttribute("src"))
    .toContain(".jpg");
});

it("matches snapshot", function () {
  const { container } = render(<Coin
    caption="Heads"
    src={HEADS}
  />);
  expect(container).toMatchSnapshot();
});