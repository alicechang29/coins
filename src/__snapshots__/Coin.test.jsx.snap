// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Coin"
  >
    <h4
      class="Coin-title"
    >
      Heads
    </h4>
    <img
      alt="Heads"
      class="Coin-image"
      src="https://coinsofamerica.com/wp-content/uploads/state-quarter-d-mint.jpg"
    />
  </div>
</div>
`;
